generators/3dParametricCurveDigitizer.cpp

```diff
@@ -0,0 +1,514 @@
+#include "DGtal/images/RigidTransformation3D.h"
+
+using namespace DGtal;
+
+
+/**
+ @page 3dParametricCurveDigitizer 3dParametricCurveDigitizer
+ @brief  Digitizes 3D parametric curves using DGtal library.
+ 
+
+
+ @b Usage:  3dParametricCurveDigitizer [options] --curve <curve> --param1 <double> --param2 <double> --param3 <double> --tstart <double> --tend <double> --step <double> --output <basename>
+
+ @b Allowed @b options @b are:
+
+ @code
+  -h  [ --help ]                   Display this message
+  --param1                  First parameter e.g. a radius or a scaling factor
+  --param2                  Second parameter e.g. a radius or a scaling factor
+  --param3                  Third parameter e.g. a distance between consecutive turns or a scaling factor
+  -ts [ --tstart]                  Start time
+  -te [ --tend]                    End time
+  -s  [ --step]                    Step
+  -k  [ --knext]                   K_NEXT value
+  -l  [ --list ]                   List all available curves
+  -c  [ --curve]                   A curve to digitize
+  -a [--angle],                    Rotation angle in radians
+  -ox                              X coordinate of origin
+  -oy                              Y coordinate of origin
+  -oz                              Z coordinate of origin
+  -ax                              X component of rotation axis
+  -ay                              Y component of rotation axis
+  -az                              Z component of rotation axis
+  -o  [ --output ] arg             Basename of the output file
+
+ @endcode
+ You can list the potential curves:
+ @code
+ $ 3dParametricCurveDigitizer --list
+ 3D Parametric curves:
+	EllipticHelix Helix with two axes and a distance between consecutive turns. Its period is 2 * Pi.
+		Required parameter(s): --param1, --param2, --param3
+	Knot_3_1 Parametric knot 3_1 i.e., a trefoli polynomial knot
+		Required parameter(s): --param1, --param2, --param3
+	Knot_3_2 Parametric knot 3_2 i.e., a trefoli knot having a period of 2 * Pi
+		Required parameter(s): --param1, --param2, --param3
+	Knot_4_1 Parametric knot 4_1 (polynomial knot)
+		Required parameter(s): --param1, --param2, --param3
+	Knot_4_3 Parametric knot 4_3 having a period 2 * Pi.
+		Required parameter(s): --param1, --param2, --param3
+	Knot_5_1 Parametric knot 5_1 (polynomial knot)
+		Required parameter(s): --param1, --param2, --param3
+	Knot_5_2 Parametric knot 5_2 having a period 2 * Pi
+		Required parameter(s): --param1, --param2, --param3
+	Knot_6_2 Parametric knot 6_2 (polynomial knot)
+		Required parameter(s): --param1, --param2, --param3
+	Knot_7_4 Parametric knot 7_4 (polynomial knot)
+		Required parameter(s): --param1, --param2, --param3
+ @endcode
+
+ @b Example:
+ @code
+ # generate a trefoli knot:
+ 3dParametricCurveDigitizer --curve Knot_3_1 --param1 10 --param2 10 --param3 10 --tstart -2.2 --tend 2.2 --step 0.001 --output knot_3_1
```

Comment:
I added the example also in the command line help to have a direct way to check the tool.

Reply:
cool, thanks